docs/design/coreclr/jit/GC-write-barriers.md

```diff
@@ -0,0 +1,79 @@
+# GC write barriers
+
+The GC write barrier function (JIT_WriteBarrier) is generally the hottest function in CoreCLR and is written in assembly. The full pseudo code for the function is as follows:
+
+
+````
+JIT_WriteBarrier(Object **dst, Object *ref)
+    Set *dst = ref
+
+    // Shadow Heap update:
+    ifdef TARGET_ARM64:
+        if *wbs_GCShadow != 0:
```

Comment:
```suggestion
        if *g_GCShadow != 0:
```
There is a mix of the wbs_ prefixes used as the local name in Arm64-specific implementation and g_ prefixes that are the global variables used throughout the runtime. I think this pseudo code should use the global variable names.